auth/src/main/java/org/openedx/auth/presentation/signin/SignInFragment.kt

```diff
@@ -197,7 +203,21 @@ private fun LoginScreen(
             uiMessage = uiMessage,
             scaffoldState = scaffoldState
         )
-
+        if (BuildConfig.PRE_LOGIN_EXPERIENCE_ENABLED) {
```

Comment:
Hey @k1rill, we can do that but in my opinion, are there any benefits to taking such a long route for a single flag? is there inspiration or reference available?

Reply:
@omerhabib26  we are trying to build Clean architecture with SOLID principles. UI shouldn't work with logic directly. Config can't be changed runtime at this moment, so we should init it once on the ViewModel init. Also when the new configuration will be merged, we have to update this code, it's better to update it once inside viewModel, instead of searching all entries. Much easier to make this route long while development, than to change it a lot of times while maintaining the code

Reply:
updated